PythonCodeConventions.md

```diff
@@ -5,16 +5,52 @@ PEP8 is the code style guide for Python code used by the main python distributio
 This style guide can be found at:
 https://www.python.org/dev/peps/pep-0008/
 
-Our coding conventions follow this style guide, with the following exceptions:
-* Always spaces. While PEP8 allows tabs for backwards compatibility, we do not.
-* Strings are double quotes. While python allows single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
-* "Maximum line length" - PEP8 specifies a maximum line length of 79 characters. At Ultimaker we have no official line length limit. And the general guideline is that the line should fit on your screen.
+Our coding conventions follow the PEP style guide, with the exceptions stated in the generic code_conventions.md provided in this repository. This also means that for rules regarding class naming, documentation, etc., the rules described in the general code convention document MUST be used.
+
+For convenience, these are re-listed here. In case of conflict, the generic code_conventions are considered to be leading.
+* Always use spaces. While PEP8 allows tabs for backwards compatibility, we do not.
+* Strings are double quotes. While Python allows single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
+* "Maximum line length" - PEP8 specifies a maximum line length of 79 characters. At Ultimaker we have no official line length limit, although we strive to keep lines below 120 characters.
 * Naming convention:
   * Function names in PEP8 are in lower_case_with_underscores. At Ultimaker we decided to match our C++ coding convention and use mixedCase (CapitalizedWords by initial lowercase character).
   * Parameters, variables and members follow the lower_case_with_underscores convention. PEP8 has no recommendation on this.
 
+As per Ultimaker RFC-1, we require that all new and changed code is typed by using mypy typing.
+
+## Examples
+### Bad example
+Normally, Python is completely dynamicly typed;
+``` python
+def fib(n):
```

Comment:
Can we please use a more readable example? This one is really bad. In fact, it does not even produce the expected result as it return a generator and not an actual value.

Reply:
Hence; bad example.